Source/WebCore/platform/mac/ScrollbarThemeMac.mm

```diff
@@ -557,9 +557,21 @@ static void scrollerImpPaint(NSScrollerImp *scrollerImp, bool enabled)
 
     GraphicsContextStateSaver stateSaver(context);
     context.clip(damageRect);
-    context.translate(scrollbar.frameRect().location());
-    LocalCurrentGraphicsContext localContext(context);
-    scrollerImpPaint(scrollbarMap().get(&scrollbar).get(), scrollbar.enabled());
+
+    auto scrollbarRect = scrollbar.frameRect();
+    if (context.platformContext()) {
+        context.translate(scrollbarRect.location());
+        LocalCurrentGraphicsContext localContext { context };
+        scrollerImpPaint(scrollbarMap().get(&scrollbar).get(), scrollbar.enabled());
+    } else {
+        auto imageBuffer = [&] {
+            auto buffer = context.createImageBuffer(scrollbarRect.size(), scrollbar.deviceScaleFactor(), DestinationColorSpace::SRGB(), context.renderingMode(), RenderingMethod::Local);
+            LocalCurrentGraphicsContext localContext { buffer->context() };
+            scrollerImpPaint(scrollbarMap().get(&scrollbar).get(), scrollbar.enabled());
```

Comment:
This could probably be factored into a lambda to share code with 564/565, but not particularly necessary. Just throwing the idea out there.

Reply:
Good point!

Reply:
Actually if you pass in a `GraphicsContext`, you could also share the `LocalCurrentGraphicsContext` code.